beacon_node/beacon_chain/src/test_utils.rs

```diff
@@ -866,14 +866,14 @@ where
             panic!("Should always be a full payload response");
         };
 
-        let signed_block = block_response.block.sign(
+        let signed_block = Arc::new(block_response.block.sign(
```

Comment:
is this test only code? If so do we also have stack overflows during tests?
(Question applies to all the uses in this file)

Reply:
yeah this is just for tests, I ended up having to make quite a few changes to make the types consistent. There's a bit of inefficient Arc-ing and de-Arc-ing, but it's just in tests so it shouldn't be too bad

Jimmy got some stack overflows in tests, but only in debug mode